### PR TITLE
Consolidate Interlocked initialize implementations

### PR DESCRIPTION
**Pull request by sharwell**
* Merge `LazyInitialization` into `InterlockedOperations` (`RoslynLazyInitializer` was not appropriate as that type is a mirror of `LazyInitializer` for nullable reference types support)
* Update implementations for consistency (initialization methods all named `Initialize`) and clarity (helper method named `GetOrStore`)

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@sharwell Consider using a more meaningful title for the PR.

**Review by jaredpar — APPROVED** [review]

File: src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs [hunk 1]
```diff
@@ -1549,7 +1549,7 @@ internal override ObsoleteAttributeData ObsoleteAttributeData
                 {
                     var result = uncommonFields._lazyObsoleteAttributeData;
                     return ReferenceEquals(result, ObsoleteAttributeData.Uninitialized)
-                        ? InterlockedOperations.Initialize(ref uncommonFields._lazyObsoleteAttributeData, null, ObsoleteAttributeData.Uninitialized)
+                        ? InterlockedOperations.Initialize(ref uncommonFields._lazyObsoleteAttributeData, initializedValue: null, ObsoleteAttributeData.Uninitialized)
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs:1552]: This may overlap/conflict with changes in #68930.

**Review by RikkiGibson — APPROVED** [review]

File: src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs (hunk 1 above)
Comment (sharwell) [reply src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs:1552]: I couldn't find any overlap/conflict

Comment (RikkiGibson) [reply src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs:1552]: I think there actually isn't any. I think the linked PR is using Interlocked.CompareExchange, not any InterlockedOperations or similar APIs that changed in this PR.

**Pull request merged**